examples/benchmark/deserialize_benchmark.py

```diff
@@ -0,0 +1,172 @@
+import torch
+import shutil
+import os
+import time
+import subprocess
+from pathlib import Path
+from tensorizer import TensorDeserializer
+from tensorizer.utils import convert_bytes, get_mem_usage
+import numpy as np
+
+# disable missing keys and unexpected key warnings
+os.environ["TRANSFORMERS_VERBOSITY"] = "error"
+
+# Improve safetensors performance
+os.environ["SAFETENSORS_FAST_GPU"] = "1"
+
+from accelerate import init_empty_weights
+from transformers import AutoModelForCausalLM, AutoTokenizer, AutoConfig
+
+
+MODEL_PATH = Path(os.environ.get("MODEL_PATH", f"./models"))
+TZR_PATH = MODEL_PATH / "model.tensors"
+HF_PATH = MODEL_PATH / "hf"
+RES_PATH = Path(os.environ.get("RES_PATH", "./results"))
+NUM_TRIALS = int(os.environ.get("NUM_TRIALS", 1))
+SKIP_HF = os.environ.get("SKIP_HF", False)
+SKIP_TZR = os.environ.get("SKIP_TZR", False)
+SKIP_ST = os.environ.get("SKIP_ST", False)
+SKIP_INFERENCE = os.environ.get("SKIP_INFERENCE", False)
+CURL_PATH = shutil.which("curl")
+
+
+RES_PATH.mkdir(parents=True, exist_ok=True)
+config = AutoConfig.from_pretrained(HF_PATH)
+
+DEVICE = torch.device("cuda:0")
+
+
+def run_curl(output: str = "/dev/null"):
+    """
+    Used to compare pod's network speed to tensorizer's speed when using S3/HTTP
+    """
+
+    http_uri = "http://tensorized.accel-object.ord1.coreweave.com/EleutherAI/gpt-j-6B/fp16/model.tensors"
+    cmd = [CURL_PATH,
+           "--header",
+           "Accept-Encoding: identity",
+           http_uri,
+           "-o",
+           output]
+
+    start = time.time()
+    subprocess.call(cmd)
+    duration = time.time() - start
+    print("CURL duration:", duration)
+
+
+def run_inference(model):
+    if SKIP_INFERENCE:
+        return
+    # Tokenize and generate
+    model.eval()
+    tokenizer = AutoTokenizer.from_pretrained(MODEL_PATH / "hf")
+    input_ids = tokenizer.encode(
+        "We the people, of the United States of America", return_tensors="pt"
+    ).to(DEVICE)
+
+    with torch.no_grad():
+        output = model.generate(input_ids, max_new_tokens=50, do_sample=True)
```

Comment:
The sequences generated here have nondeterministic lengths depending on *if* or *where* an EOS token is generated. Generating a shorter sequence some fraction of the time probably causes this step to run faster or slower at random. It doesn't affect the validity of the results of the benchmark because the randomness isn't biased in favour of any of the libraries—they should all be affected equally—but for a more consistent and precise measurement this should probably have a seed set so that it always produces the exact same sequence.
```suggestion
    torch.manual_seed(100)
    with torch.no_grad():
        output = model.generate(input_ids, max_new_tokens=50, do_sample=True)
```

Reply:
I agree and will make the change.

However, the benchmark timing doesn't include running inference so it doesn't impact the results.